core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java

```diff
@@ -0,0 +1,866 @@
+                nextType.setName(rawTestValueSource[pos + 1]);
+                MainType nextTypeMainType = new MainType();
+                nextTypeMainType.setName(rawTestValueSource[pos + 2]);
+                nextType.setMainType(nextTypeMainType);
+                nextType.setColor(rawTestValueSource[pos + 3]);
+                String parentCode = rawTestValueSource[pos + 4];
+                if (parentCode.trim().length() > 0) {
+                    nextType.setParent(parentCode);
+                }
+                expectedTumorTypeMap.put(rawTestValueSource[pos], nextType);
+            }
+        }
+    }
+
+    private void setupForTest() throws Exception {
+        setupMockRepository();
+        setupTumorTypesUtil();
+        setupExpectedTumorTypeMap();
+    }
+
+    public String makeMismatchMessage(String oncoTreeCode, String fieldName, String gotValue, String expectedValue) {
+        return oncoTreeCode + " : mismatch in " + fieldName + " (got:" + gotValue + ") (expected:" + expectedValue + ")\n";
+    }
+
+    public Set<TumorType> simulateTumorTypeApiResponse() throws Exception {
+        Map<String, TumorType> tumorTypeMap = tumorTypesUtil.getTumorTypesByVersionFromRaw(mockVersion);
+        Set<TumorType> tumorTypeSet = tumorTypesUtil.flattenTumorTypes(tumorTypeMap, null);
+        return tumorTypeSet;
+    }
+
+    private boolean testValuesMatch(String value1, String value2) {
+        if (value1 == null || value1.trim().length() == 0) {
+            return value2 == null || value2.trim().length() == 0;
+        }
+        return value1.equals(value2);
+    }
+
+    @Test
+    public void testFullOncoTreeCodeListByCode() throws Exception {
+        setupForTest();
+        Set<TumorType> tumorTypesAsPreparedForTumorTypesEndpoint = simulateTumorTypeApiResponse();
+        HashMap<String,TumorType> returnedTumorTypeMap = new HashMap<>(tumorTypesAsPreparedForTumorTypesEndpoint.size());
+        for (TumorType tumorType : tumorTypesAsPreparedForTumorTypesEndpoint) {
+            returnedTumorTypeMap.put(tumorType.getCode(), tumorType);
+        }
+        int failureCount = 0;
+        StringBuilder failureReport  = new StringBuilder();
+        for (String oncoTreeCode : expectedTumorTypeMap.keySet()) {
+            TumorType expectedTumorType = expectedTumorTypeMap.get(oncoTreeCode);
+            TumorType returnedTumorType = returnedTumorTypeMap.get(oncoTreeCode);
+            if (expectedTumorType == null || returnedTumorType == null) {
+                if (expectedTumorType == null) {
+                    failureReport.append(oncoTreeCode + " : no values in expected tumor type hash (internal error)\n");
+                }
+                if (returnedTumorType == null) {
+                    failureReport.append(oncoTreeCode + " : tumor type expected but not returned from tumorTypesUtil\n");
+                }
+            } else {
+                if (!expectedTumorType.getCode().equals(returnedTumorType.getCode())) {
```

Comment:
I have fixed this, so that null values in expected nodes are handled properly. (whitespace only string is equivalent to null)